### PR TITLE
Snowflake: Add CHANGES clause

### PR DESCRIPTION
**Pull request by jpy-git**
### Brief summary of the change made

Adds the `CHANGES` clause to the Snowflake dialect:
https://docs.snowflake.com/en/sql-reference/constructs/changes.html

### Are there any other side effects of this change that we should be aware of?
No

### Pull Request checklist
- [X] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

File: src/sqlfluff/dialects/dialect_snowflake.py [hunk 1]
```diff
@@ -602,6 +603,51 @@ class TableAliasExpressionSegment(BaseSegment):
...
+                Bracketed(
+                    OneOf("TIMESTAMP", "OFFSET", "STATEMENT"),
+                    Ref("ParameterAssignerSegment"),
+                    Ref("ExpressionSegment"),
+                ),
+            ),
+            Sequence(
+                "BEFORE",
+                Bracketed(
+                    OneOf("STATEMENT"),
```
Comment (tunetheweb) [inline src/sqlfluff/dialects/dialect_snowflake.py:633]: Does this need a `OneOf`?

```suggestion
                   "STATEMENT",
```

**Review by tunetheweb — APPROVED** [review]
One small nit

File: src/sqlfluff/dialects/dialect_snowflake.py (hunk 1 above)
Comment (jpy-git) [reply src/sqlfluff/dialects/dialect_snowflake.py:633]: whoops, good spot 👍

File: src/sqlfluff/dialects/dialect_snowflake.py [hunk 2]
```diff
@@ -640,6 +641,51 @@ class TableAliasExpressionSegment(BaseSegment):
...
+                Bracketed(
+                    OneOf("TIMESTAMP", "OFFSET", "STATEMENT"),
+                    Ref("ParameterAssignerSegment"),
+                    Ref("ExpressionSegment"),
+                ),
+            ),
+            Sequence(
+                "BEFORE",
+                Bracketed(
+                   "STATEMENT",
```
Comment (tunetheweb) [inline src/sqlfluff/dialects/dialect_snowflake.py:671]: Oops. Missed a space.

```suggestion
                    "STATEMENT",
```

**Review by tunetheweb — APPROVED** [review]

**Pull request merged**